partiql-planner/src/main/kotlin/org/partiql/planner/internal/transforms/RelConverter.kt

```diff
@@ -441,24 +471,34 @@ internal object RelConverter {
             return rel(type, op)
         }
 
+        private fun visitIfQuerySet(expr: Expr): Rel {
+            return when (expr) {
+                is Expr.QuerySet -> visit(expr, nil)
+                else -> {
+                    val rex = RexConverter.applyRel(expr, env)
+                    val op = relOpScan(rex)
+                    val type = Rel.Type(listOf(Rel.Binding("_1", ANY)), props = emptySet())
+                    return rel(type, op)
+                }
+            }
+        }
+
         /**
          * Append SQL set operator if present
          */
-        private fun convertSetOp(input: Rel, setOp: Expr.SFW.SetOp?): Rel {
-            if (setOp == null) {
-                return input
-            }
-            val type = input.type.copy(props = emptySet())
-            val lhs = input
-            val rhs = visitExprSFW(setOp.operand, nil)
-            val quantifier = when (setOp.type.setq) {
-                SetQuantifier.ALL -> Rel.Op.Set.Quantifier.ALL
-                null, SetQuantifier.DISTINCT -> Rel.Op.Set.Quantifier.DISTINCT
+        private fun convertSetOp(setExpr: QueryBody.SetOp): Rel {
```

Comment:
(self-review) one benefit of reusing the SQL set op plan nodes for the PartiQL bag ops is that we can reuse the `Rel` scan behavior for non-collections (i.e. coercing to a bag in permissive and erroring in strict mode).